src/filesystem/windows/SDL_sysfilesystem.c

```diff
@@ -373,4 +375,101 @@ char *SDL_SYS_GetCurrentDirectory(void)
     return retval;
 }
 
+SDL_IOStream *SDL_SYS_CreateSafeTempFile(void)
+{
+    wchar_t tmp_folder[MAX_PATH];
+    wchar_t tmp_file[MAX_PATH];
+    HANDLE hFile;
+
+    /* There exists GetTempPath2W, which is available only on Windows 11+ */
+    DWORD tmp_len = GetTempPathW(MAX_PATH, tmp_folder);
+
+    if (tmp_len == 0 || tmp_len > MAX_PATH) {
+        WIN_SetError("Couldn't get the temporary folder");
+        return NULL;
+    }
+
+    /* Note: Someone watching the temp folder could create a file with a
+       clashing name before it is re-created (TOCTOU). However, this shouldn't
```

Comment:
I don't know the security implications of this on Windows. Is the temp directory per-user (good), or is it shared like `/tmp` is (bad, unless carefully mitigated)?

Someone who knows Windows' security properties should review this.

Reply:
The temporary folder path is something like `C:\Users\User\AppData\Roaming\Temp\` -- the exact path might be slightly different, but it's within the current user's home directory, so presumably they are separate for each user. I think even the system has its own temporary folder.

Reply:
If filesystem permissions prevent a (non-privileged) user from accessing a different user's temp - e.g. `C:\Users\Alice\AppData\Roaming\Temp` is inaccessible by Bob - then it isn't necessary to be as paranoid as when dealing with Unix `/tmp`.